src/server/parser/parser.ts

```diff
@@ -486,7 +485,7 @@ export class Parser
 		if (!leftSquare)
 			return Err('UnreachableState')
 		node.add(leftSquare)
-		let index: ASTNode | undefined
+		let index: ASTNode | undefined = undefined
```

Comment:
Does this still have to be explicitly assigned undefined, or can the implicit undefined of `let` be used?

Reply:
It can be implicit. That was a mistake on my part. Good catch. Reverted the change